app/ember/src/client/preview/render.js

```diff
@@ -0,0 +1,91 @@
+/* eslint-disable no-undef */
+import { window, document } from 'global';
+import { stripIndents } from 'common-tags';
+import { getApplicationConfig } from './utils';
+
+const config = getApplicationConfig();
+
+const rootEl = document.getElementById('root');
+
+const app = window.require(`${config.APP.name}/app`).default.create({
+  autoboot: false,
+  rootElement: rootEl,
+});
+
+let appInstance;
+
+function render(options, el) {
+  const { template, context = {}, boundingDocument } = options;
+
+  if (!appInstance) {
+    app
+      .boot()
+      .then(() => {
+        const appInstancePrivate = app.buildInstance();
+        return appInstancePrivate.boot().then(() => {
+          appInstance = appInstancePrivate;
+
+          return appInstance;
+        });
+      })
+      .then(builtAppInstance => {
+        builtAppInstance.register(
+          'component:story-mode',
+          Ember.Component.extend({
+            layout: template || options,
+            ...context,
+          })
+        );
+        const component = builtAppInstance.lookup('component:story-mode');
+
+        if (boundingDocument) {
+          component.appendTo(boundingDocument.querySelector('#component'));
+
+          el.append(boundingDocument);
+        } else {
+          component.appendTo(el);
+        }
+      });
+  } else {
+    $(appInstance.rootElement).empty();
```

Comment:
We shouldn't need to manually empty like this, we should call `Ember.run(component, 'destroy')` on the component we created in the first pass. Destroying the component will teardown its DOM (calling any lifecycle hooks on the way).

Reply:
talked to you about the solution and pushed the latest